src/rkopenmdao/runge_kutta_scheme.py

```diff
@@ -32,13 +48,18 @@ def __init__(
         ],
         # stage_perturbation, stage_time, delta_t, butcher_diagonal_element
         # -> old_state_perturbation, accumulated_stages_perturbation
+        error_controller: ErrorController = None
```

Comment:
Seeing as you define a default later, you can use that default here instead of `None`. If you want to control with that whether to do adaptive time stepping or not, I would instead add a flag (e.g. `use_adaptive_time_stepping: bool = False`). Then you can easily have varied behavior:

- If the flag is `False`, time stepping is done as usual, ignoring the error controller and the embedded scheme should the passed butcher tableau have one.
- If the flag is `True`, you can use adaptive time stepping with the passed error controller (or the default one, if not explicitely specified, provided the butcher tableau has an embedded scheme. If not, you'd throw an error in that case.

Reply:
While we discussed that you can't use a default here due to the dependence of the `ErrorController` on the order of the scheme, the point of controlling whether to use adaptive time stepping or not with a flag still stands.